migration guide: add section for documenting cylc8 changes (#241)
cylc/cylc-doc: src/7-to-8/caveats.rst

Cylc |version| Caveats
======================

> This is an early release of Cylc, there are some loose ends and features which

Reviewer: ```suggestion
This is an early release of Cylc 8, there are some loose ends and features which
```

have not yet been (re)implemented.

The documentation has not been fully reviewed and command line information may
be out of date in some cases.

Cylc Flow
---------

Run Numbers
   By default when you install a workflow with ``cylc install`` it is installed
   into a numbered subdirectory e.g. ``run1``. A subsequent installation
   would go into ``run2``, etc.

   At present the full workflow registration *including* this run number must
   be provided to Cylc commands e.g::

      $ cylc play myflow/run1

   We plan to infer the most recent run to make this easier e.g::

      $ cylc play myflow

   * https://github.com/cylc/cylc-flow/issues/3895
CLI Change
   We plan to move to a new way of specifying workflow, cycles, tasks and jobs
   on the command line (with back support for the old format).

>    This will open up performing operations on multiple workflows in a single

Reviewer: ```suggestion
   This will allow performing operations on multiple workflows in a single
```

   command e.g::

      $ cylc stop '*'  # stop everything

   * https://github.com/cylc/cylc-flow/pull/3931   
Platform Selection
   When Cylc needs to perform an action on a platform (e.g. submit a job)
   it picks a random host from the platform. If this host is down the operation
   will fail.

   In a later relase Cylc will pick another host in order to provide resilience
   to system issues.

   * https://github.com/cylc/cylc-flow/issues/3827
Trigger Edit
   Functionality removed pending reimplementation.

   * https://github.com/cylc/cylc-flow/issues/3751
Hold / Release

>    The ability to hold tasks is in place, however, doesn't work for most uses.

Reviewer: ```suggestion
   The ability to hold specific tasks is in place, however, it doesn't work for most uses with the new scheduling algorithm.
```
